src/diffusers/pipelines/ddim/pipeline_ddim.py

```diff
@@ -105,11 +105,11 @@ def __call__(
             image = self.scheduler.step(
                 model_output, t, image, eta=eta, use_clipped_model_output=use_clipped_model_output, generator=generator
             ).prev_sample
-
-        image = (image / 2 + 0.5).clamp(0, 1)
-        image = image.cpu().permute(0, 2, 3, 1).numpy()
-        if output_type == "pil":
-            image = self.numpy_to_pil(image)
+        if output_type != "latent":
+            image = (image / 2 + 0.5).clamp(0, 1)
+            image = image.cpu().permute(0, 2, 3, 1).numpy()
+            if output_type == "pil":
+                image = self.numpy_to_pil(image)
```

Comment:
Everywhere we check output_type, can we please check it like

```py
if output_type == "<>":
  ...
elif output_type == "<>":
  ...
elif output_type == "<>:
  ...
```

with no `!=` and no nested conditionals

This will make it easier to check the logic is correct and to merge the pr :)